### PR TITLE
Local: Add GP_SCRIPT_DEBUG to load the unminified CSS and JS

### PR DESCRIPTION
**Pull request by amieiro**
## What?

This PR improves the workflow for the unminified CSS and JS.

Resolves https://github.com/GlotPress/GlotPress/issues/1587.

## Why?

This PR resolves 2 problems:
* When a developer updates CSS or JS to a PR, she needs to add the minified files to the PR, so the reviewer can review the PR.
* When you clone the repository, you need to know that for loading the unminified assets (CSS and JS), you have to activate the SCRIPT_DEBUG constant in the wp-config.php file, so if you want to improve some CSS or JS, you have to activate this flag.

## How?

This PR proposes to add a new constant `GP_SCRIPT_DEBUG`, set to true by default, so:
* The new PRs don't need to commit the minified files.
* The users that clone the repository will use the unminified files by default.

When we will create a new release, we will need to add 2 steps to the release process:
* change this new constant `GP_SCRIPT_DEBUG` to false.
* MInify the assets with the Grunt task: `npm run build`.

This PR adds 2 new Grunt actions:
* the `browserify` action, to create the inline-translation.js file.
* the `copy` action, to copy the CSS and JavaScript files to the assets' folder (`assets/css` and `assets/js`).

## Testing Instructions

1. Execute `npm update` to add the new dependencies.
1. Execute the `npm run build` task to get the minified versions and to copy the JS and CSS files to the new destinations.
1. Set `SCRIPT_DEBUG` (wp-config.php file) and `GP_SCRIPT_DEBUG` (glotpress.php) to `false`, and check that you receive the minified files.
1. Set `SCRIPT_DEBUG` to `true` and `GP_SCRIPT_DEBUG` to `false`, and check that you receive the unminified files.
1. Set `SCRIPT_DEBUG` to `false` and `GP_SCRIPT_DEBUG` to `true`, and check that you receive the unminified files.
1. Set `SCRIPT_DEBUG` to `true` and `GP_SCRIPT_DEBUG` to `true`, and check that you receive the unminified files.

Please, check you have these files:

* Minified version (WP backend):
    * assets/css/inline-translation-loader.min.css
    * assets/css/inline-translation.min.css
    * assets/js/inline-translation-loader.min.js
    * assets/js/inline-translation.min.js
    
* Minified version (translation backend):
    * assets/css/style.min.css
    * assets/js/vendor/jquery.tablesorter.min.js
    * assets/js/common.min.js
    * assets/js/editor.min.js

* Unminified version (WP backend):
    * assets/css/inline-translation-loader.css
    * assets/css/inline-translation.css
    * assets/js/inline-translation-loader.js
    * inline-translation/inline-translation.js

* Unminified version (translation backend):
    * assets/css/style.css
    * assets/js/vendor/jquery.tablesorter.js
    * assets/js/common.js
    * assets/js/editor.js

### CODE REVIEW

**Comment by pedro-mendonca** [comment]
This might be usefull globally and added to the `develop` branch.

**Comment by amieiro** [comment]
> This might be usefull globally and added to the `develop` branch.

Yes, of course, the idea is to add this PR to the `develop` branch, but we want to use it with the all the JavaScript files, including the new ones in the `local` branch.

**Review by akirk — APPROVED** [review]
I removed the copy module as I don't see it necessary (basically now the concat module is doing its job for CSS and the browserify for JS). I've tested it and it works fine for me now, both the watch and the build. Nice!

**Comment by akirk** [comment]
I've done some more changes to keep the PR minimal. One thing also to test is `npm run watch` and then save a file in the inline-translator JS or CSS. It will now kick off browserify or concat.

**Comment by amieiro** [comment]
> I removed the copy module as I don't see it necessary (basically now the concat module is doing its job for CSS and the browserify for JS). I've tested it and it works fine for me now, both the watch and the build. Nice!

The copy action is needed to copy the `inline-translation.min.js` from the `inline-translation` folder to the `assets\js` folder.

**Comment by akirk** [comment]
Ah, I made a mistake resolving the conflict. Let me commit it right. Browserify should just create the unminified file in the assets directory.

File: gp-includes/inline-translation.php [hunk 1]
```diff
@@ -131,11 +131,10 @@ public function __construct() {
 	public function enqueue_scripts() {
 		$css_extension = SCRIPT_DEBUG || GP_SCRIPT_DEBUG ? '.css' : '.min.css';
 		$js_extension = SCRIPT_DEBUG || GP_SCRIPT_DEBUG ? '.js' : '.min.js';
-		$inline_translation_js_file = SCRIPT_DEBUG || GP_SCRIPT_DEBUG ? 'inline-translation/inline-translation.js' : 'assets/js/inline-translation.min.js';
 
 		wp_enqueue_style( 'inline-translation-loader', gp_plugin_url( 'assets/css/inline-translation-loader' . $css_extension, __FILE__ ) ); // phpcs:ignore WordPress.WP.EnqueuedResourceParameters.MissingVersion
 		wp_enqueue_style( 'inline-translation', gp_plugin_url( 'assets/css/inline-translation' . $css_extension, __FILE__ ) ); // phpcs:ignore WordPress.WP.EnqueuedResourceParameters.MissingVersion
-		wp_enqueue_script( 'inline-translation', gp_plugin_url( $inline_translation_js_file ), array( 'jquery', 'jquery-ui-tooltip' ), false, true ); // phpcs:ignore WordPress.WP.EnqueuedResourceParameters.NoExplicitVersion
+		wp_enqueue_script( 'inline-translation', gp_plugin_url( 'assets/js/inline-translation' . $js_extension ), array( 'jquery', 'jquery-ui-tooltip' ), false, true ); // phpcs:ignore WordPress.WP.EnqueuedResourceParameters.NoExplicitVersion
```
Comment (akirk) [inline gp-includes/inline-translation.php:137]: Oh, thanks!

**Comment by amieiro** [comment]
@akirk can I delete the `inline-translation\Makefile` file, because now its actions are made by Grunt, and then merge this branch into `local` branch?

**Review by akirk — APPROVED** [review]
Ok to merge

**Pull request merged**